transport/grpc/outbound_test.go

```diff
@@ -50,6 +50,32 @@ func TestNoRequest(t *testing.T) {
 	assert.Equal(t, yarpcerrors.InvalidArgumentErrorf("request for grpc outbound was nil"), err)
 }
 
+func TestCallWithInvalidHeaderValue(t *testing.T) {
+	listener, err := net.Listen("tcp", "127.0.0.1:0")
+	require.NoError(t, err)
+
+	tran := NewTransport()
+	out := tran.NewSingleOutbound(listener.Addr().String())
+	require.NoError(t, tran.Start())
+	require.NoError(t, out.Start())
+	defer tran.Stop()
+	defer out.Stop()
+
+	ctx, cancel := context.WithTimeout(context.Background(), time.Millisecond*100)
+	defer cancel()
+	req := &transport.Request{
+		Caller:    "caller",
+		Service:   "service",
+		Encoding:  transport.Encoding("raw"),
+		Procedure: "proc",
+		Headers: transport.NewHeaders().With("valid-key", `value with
+line feed`),
+	}
+	_, err = out.Call(ctx, req)
+
+	require.Contains(t, err.Error(), yarpcerrors.InvalidArgumentErrorf("grpc request header value contains invalid characters including ASCII 0xd, 0xa, or 0x0").Error())
```

Comment:
Can you enhance the test and also try the two other cases please?

Reply:
Sure